packages/merge-styles/src/concatStyleSets.ts

```diff
@@ -17,7 +18,7 @@ export function concatStyleSets<TStyleSet>(
  * @param styleSet2 - The second style set to be concatenated.
  */
 export function concatStyleSets<TStyleSet1, TStyleSet2>(
-  styleSet1: TStyleSet1 | false | null | undefined,
+  styleSet1: TStyleSet1 | false | null | undefined | ShadowConfig,
```

Comment:
Do you need this change just for the first one in each of these declarations?

Reply:
Yes, this allows users to optionally pass a `ShadowConfig` in as the first option to control the shadow DOM behavior while allowing the implementation to only have to check one place for this feature which is a performance optimization. I don't pass it in last because many `merge-styles` APIs use `...args` which complicates reading this config.